yt_dlp/extractor/tvp.py

```diff
@@ -449,57 +481,101 @@ def _real_extract(self, url):
+            'thumbnail': 're:https://.+',
         },
-        'params': {
-            'skip_download': True,
-        },
-        'add_ie': ['TVPEmbed'],
-    }, {
-        'url': 'https://vod.tvp.pl/website/lzy-cennet,38678312',
-        'only_matching': True,
     }]
 
-    def _entries(self, display_id, playlist_id):
-        url = 'https://vod.tvp.pl/website/%s,%s/video' % (display_id, playlist_id)
-        for page_num in itertools.count(1):
-            page = self._download_webpage(
-                url, display_id, 'Downloading page %d' % page_num,
-                query={'page': page_num})
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
 
-            video_ids = orderedSet(re.findall(
-                r'<a[^>]+\bhref=["\']/video/%s,[^,]+,(\d+)' % display_id,
-                page))
+        return self._parse_video(self._api_request(f'vods/{video_id}', video_id))
 
-            if not video_ids:
-                break
 
-            for video_id in video_ids:
-                yield self.url_result(
-                    'tvp:%s' % video_id, ie=TVPEmbedIE.ie_key(),
-                    video_id=video_id)
+class TVPVODSeriesIE(TVPVODBaseIE):
+    IE_NAME = 'tvp:vod:series'
+    _VALID_URL = r'https?://vod\.tvp\.pl/[a-z\d-]+,\d+/[a-z\d-]+-odcinki,(?P<id>\d+)(?:\?[^#]+)?(?:#.+)?$'
+
+    _TESTS = [{
+        'url': 'https://vod.tvp.pl/seriale,18/ranczo-odcinki,316445',
+        'info_dict': {
+            'id': '316445',
+        },
+        'playlist_count': 312,
+    }, {
+        'url': 'https://vod.tvp.pl/programy,88/rolnik-szuka-zony-odcinki,284514',
+        'only_matching': True,
+    }, {
+        'url': 'https://vod.tvp.pl/dla-dzieci,24/laboratorium-alchemika-odcinki,309338',
+        'only_matching': True,
+    }]
+
+    def _entries(self, seasons, playlist_id):
+        for season in seasons:
+            episodes = self._api_request(
+                f'vods/serials/{playlist_id}/seasons/{season["id"]}/episodes', playlist_id,
+                note=f'Downloading episode list for {season["title"]}')
+            yield from (self._parse_video(episode) for episode in episodes)
 
     def _real_extract(self, url):
-        mobj = self._match_valid_url(url)
-        display_id, playlist_id = mobj.group('display_id', 'id')
-        return self.playlist_result(
-            self._entries(display_id, playlist_id), playlist_id)
+        playlist_id = self._match_id(url)
+        metadata = self._api_request(
+            f'vods/serials/{playlist_id}', playlist_id,
+            note='Downloading serial metadata')
+        seasons = self._api_request(
+            f'vods/serials/{playlist_id}/seasons', playlist_id,
+            note='Downloading season list')
+        return {
+            '_type': 'playlist',
+            'title': metadata['title'],
+            'description': clean_html(traverse_obj(metadata, ('description', 'lead'))),
+            'entries': self._entries(seasons, playlist_id),
+        }
```

Comment:
`title` is not mandatory in a playlist, but there should be an ID. Also, could use `playlist_result()` method; consider extracting `metadata['mainCategory']['name']` as a category.
```suggestion
        return self.playlist_result(
            self._entries(seasons, playlist_id),
            playlist_id, metadata.get('title'),
            clean_html(traverse_obj(metadata, ('description', 'lead'))))
```